core/src/main/java/fj/Semigroup.java

```diff
@@ -97,6 +98,27 @@ public A multiply1p(int n, A a) {
   }
 
   /**
+   * Sums the given values with right-fold.
+   */
+  public A sumRight(final NonEmptyList<A> as) {
```

Comment:
what about naming it ```sumNel```? associativity make sumRight and sumLeft equivalent, and thus using foldLeft1 would be more performant...